### PR TITLE
[backport] ROCm: Fix filters in `cupyx.scipy.ndimage` - Part 1

### PR DESCRIPTION
**Pull request by kmaehashi**
Backport of #4271

### CODE REVIEW

**Comment by kmaehashi** [comment]
[automatic post] Jenkins, test this please.

File: cupy/core/include/cupy/math_constants.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#ifndef CUPY_MATH_CONSTANTS_H
```
Comment (grlee77) [inline cupy/core/include/cupy/math_constants.h:1]: The backport should probably use the more minimal version of this file that was introduced for #4565. 

#4565 also helped avoid a patch during the conda package build

Comment (emcastillo) [reply cupy/core/include/cupy/math_constants.h:1]: Yeah, we need to backport that one first :)

Comment (leofang) [reply cupy/core/include/cupy/math_constants.h:1]: Good catch, @grlee77!

Comment (leofang) [reply cupy/core/include/cupy/math_constants.h:1]: Perhaps this PR can be backported first, and then followed by backporting #4565?

**Comment by emcastillo** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]

**Pull request merged**